ci: add CodeQL analysis workflow

Brijeshthummar02 opened the pull request:
Fixes #524

brenoepics commented:
Good Job, LGTM, I just updated some of the GA's and changed the preferred style to pin to the commit hash, thanks for your contribution!

Brijeshthummar02 commented:
> Good Job, LGTM, I just updated some of the GA's and changed the preferred style to pin to the commit hash, thanks for your contribution!

No worries! My pleasure.

Brijeshthummar02 commented:
> Good Job, LGTM, I just updated some of the GA's and changed the preferred style to pin to the commit hash, thanks for your contribution!

Further while when it gets activated on other New Contributors PR and if any errors exist i will tackle it down, don't worry!
